lib/video_merger/buffer_queue.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Membrane.VideoMerger.BufferQueue do
+  @moduledoc false
+
+  alias Membrane.{Buffer, Pad}
+
+  @type k :: Pad.dynamic_id_t()
+  @type v :: Buffer.t()
+  @type t :: map
+
+  @eos %Buffer{metadata: %{pts: -1}, payload: :end_of_stream}
+
+  @doc """
+  Creates an empty queue.
+  """
+  @spec new() :: t
+  def new(), do: %{}
+
+  @doc """
+  Returns list of ids with empty queues.
+  """
+  @spec get_empty_ids(t) :: [k]
+  def get_empty_ids(state) do
+    state
+    |> Enum.filter(fn {_id, buffers} -> buffers == [] end)
+    |> Enum.map(fn {id, _val} -> id end)
+  end
+
+  @doc """
+  Enqueues list of buffers to queue of given id.
+  """
+  @spec enqueue_list(t, k, [v]) :: t
+  def enqueue_list(queue, id, buffers), do: Map.update(queue, id, [buffers], &(&1 ++ buffers))
```

Comment:
I started to think about performance as in general `++` shouldn't be often used as it will have linear complexity on size of the LHS list, but I guess it might require using some kind of library for double ended queue and should be left for the next phase - thus you may create an issue for that